src/SecureElementReader.App/ViewModels/VerifyPinDialogViewModel.cs

```diff
@@ -84,71 +120,94 @@ private void ShowErrorMsg(List<string> errorList)
+                        ContentHeader = _mainWindowProvider.GetMainWindow().GetResourceObservable("Error").ToString(),
+                        ContentTitle = _mainWindowProvider.GetMainWindow().GetResourceObservable("Error").ToString(),
                         ShowInCenter = true,
                         Icon = MessageBoxAvaloniaEnums.Icon.Error,
                         Topmost = true,
                         WindowStartupLocation = Avalonia.Controls.WindowStartupLocation.CenterOwner,
                         SizeToContent = Avalonia.Controls.SizeToContent.WidthAndHeight
                     });
-            msg.ShowDialog(mainWindowProvider.GetMainWindow());
+            msg.ShowDialog(_mainWindowProvider.GetMainWindow());
         }
 
         private void ValidateResult(VerifyPinModel result)
         {
             if (result.PkiPinSuccess)
             {
-                PkiResult = Properties.Resources.PkiPinOK;
-                PkiResultColor = Avalonia.Media.Brushes.Green;
+                PkiPinOk = true;
+                PinMustBe4Char = false;
+                PleaseInsertPin = false;
+                WrongPinAlertText = false;
             }
             else
             {
-                if (result.PKIAppletLocked)
+                if (result.PkiAppletLocked)
                 {
-                    PkiResult = Properties.Resources.PkiAppletLocked;
+                    PkiAppletLocked = true;
+                    PinMustBe4Char = false;
+                    PleaseInsertPin = false;
+                    PkiPinOk = false;
+                    SePinOk = false;
+
                 }
                 else
                 {
-                    PkiResult =  Properties.Resources.PkiPinWrong.Replace("[xxx]", (result.PkiTrysLeft - 192).ToString());
-                }                
-                PkiResultColor = Avalonia.Media.Brushes.Red;
+                    TrysLeft = (result.PkiTrysLeft - 192).ToString();
+                    WrongPinAlertText = true;
+                    PinMustBe4Char = false;
+                    PleaseInsertPin = false;
+                    SePinOk = false;
+                    PkiPinOk = false;
+                }
             }
 
             if (result.SePinSuccess)
             {
-                SeResult = Properties.Resources.SePinOK;
-                SeResultColor = Avalonia.Media.Brushes.Green;
+                SePinOk = true;
+                PinMustBe4Char = false;
+                PleaseInsertPin = false;
+                WrongPinAlertText = false;
             }
             else
             {
-                if (result.SEAppletLocked)
+                if (result.SeAppletLocked)
                 {
-                    SeResult = Properties.Resources.SeAppletLocked;
+                    SeAppletLocked = true;
+                    PinMustBe4Char = false;
+                    PleaseInsertPin = false;
+                    WrongPinAlertText = false;
+
```

Comment:
Delete empty row before curly brace {

You should install some VS extension to help you clean code like https://marketplace.visualstudio.com/items?itemName=SteveCadwallader.CodeMaid